test-programs/src/bin/client_get_authority.rs

```diff
@@ -0,0 +1,14 @@
+use waki::Client;
+
+fn main() {
+    let req = Client::new()
+        .get("https://httpbin.org/get?a=b")
+        .query(&[("c", "d")])
+        .build()
+        .unwrap();
+
+    match req.authority() {
+        Some(authority) => assert_eq!(authority.as_str(), "httpbin.org"),
+        None => assert!(false, "Authority isn't set on client-request"),
+    }
```

Comment:
I think this test case is unnecessary because it's already covered by the authority test case for the server. The other changes LGTM.

Reply:
thanks you, i have removed the unnecessary client test for authority now 👍